content/200-orm/200-prisma-client/500-deployment/301-edge/485-deploy-to-vercel.mdx

```diff
@@ -0,0 +1,458 @@
+
+```terminal
+npm install -g vercel
+npx create-next-app@latest
+npm install prisma --save-dev
+npx prisma init
+```
+
+We'll use the default `User` model for the example below:
+
+```prisma
+model User {
+  id    Int     @id @default(autoincrement())
+  email String  @unique
+  name  String?
+}
+```
+
+### Vercel Postgres
+
+If you are using Vercel Postres, you need to:
+
+- use the `@prisma/adapter-neon` database adapter (via the `driverAdapters` Preview feature) because Vercel Postgres uses [Neon](https://neon.tech/) under the hood
+- be aware that Vercel by default calls the environment variable for the database connection string `POSTGRES_PRISMA_URL` while the default name used in the Prisma docs is typically `DATABASE_URL`; using Vercel's naming, you need to set the following fields on your `datasource` block:
+  ```prisma
+  datasource db {
+    provider  = "postgresql"
+    url       = env("POSTGRES_PRISMA_URL") // uses connection pooling
+    directUrl = env("POSTGRES_URL_NON_POOLING") // uses a direct connection
+  }
+  ```
+
+#### 1. Configure Prisma schema & database connection
+
+First, ensure that database connection is configured properly. In your Prisma schema, set the `url` of the `datasource` block to the `POSTGRES_PRISMA_URL` and the `directUrl` to the `POSTGRES_URL_NON_POOLING` environment variable. You also need to enable the `driverAdapters` feature flag:
+
+```prisma file=schema.prisma
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["driverAdapters"]
+}
+
+datasource db {
+  provider  = "postgresql"
+  url       = env("POSTGRES_PRISMA_URL") // uses connection pooling
+  directUrl = env("POSTGRES_URL_NON_POOLING") // uses a direct connection
+}
+```
+
+Next, you need to set the `POSTGRES_PRISMA_URL` and `POSTGRES_URL_NON_POOLING` environment variable to the values of your database connection.
+
+If you ran `npx prisma init`, you can use the `.env` file that was created by this command to set these:
+
+```bash file=.env
+POSTGRES_PRISMA_URL="postgres://user:password@host-pooler.region.postgres.vercel-storage.com:5432/name?pgbouncer=true&connect_timeout=15"
+POSTGRES_URL_NON_POOLING="postgres://user:password@host.region.postgres.vercel-storage.com:5432/name"
+```
+
+#### 2. Install dependencies
+
+Next, install the required packages:
+
+```terminal
+npm install @prisma/adapter-neon
+npm install @neondatabase/serverless
+```
+
+#### 3. Configure `postinstall` hook
+
+Next, add a new key to the `scripts` section in your `package.json`:
+
+```js file=package.json highlight=5;add
+{
+  // ...
+  "scripts": {
+    // ...
+    "postinstall": "prisma generate"
+  }
+}
+```
```

Comment:
Wasn't this covered above?
https://docs-git-edge-support-prisma.vercel.app/orm/prisma-client/deployment/edge/deploy-to-vercel#generate-prisma-client-in-postinstall-hook

Reply:
The instructions above feel out of place to me. It is not edge function specific, same for https://docs-git-edge-support-prisma.vercel.app/orm/prisma-client/deployment/edge/deploy-to-vercel#setting-your-database-connection-url-as-an-environment-variable - so maybe just remove them above if they are repeated in each of the "mini tutorials" anyway?